rust/src/delta.rs

```diff
@@ -1034,6 +1057,30 @@ impl DeltaTable {
         })
     }
 
+    pub async fn create(
+        &mut self, 
+        metadata: DeltaTableMetaData, 
+        protocol: action::Protocol
+    ) -> Result<(), DeltaTableError> {
+        // need to check if table already exists so that protocal and metaData arent overwritten on second create
+        
+        let meta = action::MetaData::try_from(metadata).unwrap();
```

Comment:
we should propagate the error back to caller here instead of using unwrap :)

Reply:
what method would you use to propagate the error back up? I tried the `?` but I am getting compile errors from that. I have seen a lot of ways to check with like `match` and stuff, but is there a preferred way, such as `unwrap_or_else` or possibly another method?

Reply:
Yes, once the from trait for the error you want to propagate is implemented for `DeltaTableError`, you will be able to use `?`. 

We use `thiserror` crate to automatically implement error conversion for `DeltaTableError` at: https://github.com/delta-io/delta-rs/blob/main/rust/src/delta.rs#L70`

Reply:
is unwrapping the try_from conversion from DeltaTableMetaData -> MetaData actually result in an error that would be considered a DeltaTableError? Ive never used the `thiserror` create and having trouble conceptualizing where the Error needs to be implemented to allow the `?` operator to be valid, sorry

Reply:
Unwrap on a Result causes the program to crash immediately. You would want to put `?` on the `Result` itself, not the return from unwrap. So basically replace unwrap with `?`. `thiserror` is just a helper library that reduces boilerplate code to implement `From` trait for a Result's error enum variant. What the `?` operator does is it uses that `From` trait implementation to convert the Error enum variant to the error type used by the enclosing function, in this case `DeltaTableError`.